docs/design/editorial_design.md

```diff
@@ -0,0 +1,354 @@
+    item: Item,
+    track: Track,
+    composition_time: RationalTime,
+    fill_template: Item = None, # Default to Gap
+)
+```
+
+---
+
+## Trim
+![Trim](../_static/edit/03_trim.png)
+- Adjust a single item's start time or duration.
+- Do not affect other clips.
+- Fill now-empty time with gap or template.
+- Clamps source_range to non-gap boundary (can only overwrite gaps)
+
+#### API
+```py
+"""
+Args:
+    item: Item to apply trim to
+    delta_in: RationalTime that the item's source_range().start_time()
+        will be adjusted by
+    delta_out: RationalTime that the item's
+        source_range().end_time_exclusive() will be adjusted by
+"""
+otio.algorithms.trim(
+    item: Item,
+    delta_in: RationalTime = None,   #< Duration of change
+    delta_out: RationalTime = None,  #< Duration of change
+    fill_template: Item = None,      #< Default to Gap
+)
+```
+> Not convinced on the argument names for this
+
+---
+
+## Slice
+![Slice](../_static/edit/04_slice.png)
+- Slice an item, generating a clone of self and augment both item's source_range.
+
+#### API
+```py
+"""
+Args:
+    item: otio.schema.Item - to cut
+    at_time: otio.schema.RationalTime - time, based on the
+        coordinates to slice at
+    coordinates: Enumerator for the variable calculations that
+        can be done to the at_time.
+
+Example:
+    Let item = A
+    let at_time = 25 @ 24fps
+
+    0                                             N
+    | ------------------------------------------- |
+    | [0    GAP  20][0       A      50]           |
+    | ------------------------------------------- |
+
+    if coordinates == Local:
+    | ------------------------------------------- |
+    | [0    GAP  20][0    A   25][26   A  50]     |
+    | ------------------------------------------- |
+
+    if coordinates == Parent:
+    | ------------------------------------------- |
+    | [0    GAP  20][0 A 5][6   A  50]            |
+    | ------------------------------------------- |
+
+    if coordinates == Global:
+        - This only matters if the parent is a track in a compound
+          stack, at which point we might be better off having the
+          user just convert from one time to the other
+"""
+otio.algorithms.slice(
+    item: Item,
+    at_time: RationalTime,
+    coordinates: otio.algorithms.Coordinates.(Local|Parent|Global*)
+)
+```
+> I'm not sure if we have a coordinate enumerator already?
+
+---
+
+
+## Slip
+![Slip](../_static/edit/05_slip.png)
+- Adjust the start_time of an item's source_range.
+- Do not affect surrounding items.
+- Clamp to available_range of media (if available)
+
+#### API
+```py
+"""
+Args:
+    item: Item that we're slipping
+    delta: RationalTime to adjust the range by.
+"""
+otio.algorithms.slip(
+    item: Item,
+    delta: RationalTime,
+)
+```
+- In the example image, `C` is `item` and `delta` is the arrow's vector.
+
+> Effectively this is already quite simple in the core API but would be nice to have here.
```

Comment:
I suspect the value of some of these functions might be that they hide the underlying complexity of whether or not source_range is set.

Maybe you split these functions into a "maybe" and "Definitely" pile?  that will help prioritize them and see if people come out with concerns or use cases.

This might be a controversial position, but I think I'd also prefer to not have things named too close even if they do mean a specific thing (slip/slide for example)... unless that verb means the same thing in all major editing packages.